Add content/copy support to eZPlatformSearch

harmstyler opened the pull request:
This change modifies the `addObject` function to test if the content object being
added to the solr index is available in the database before continuing.
The legacy module content/copy triggers the `addObject` function before
the new content is actually in the database, causing an uncaught
exception in the search plugin. The try/catch approach seems best
because the function is not sent any information about how the
`$contentObject` was created. The copied content is then added at a later
time by a frequent cronjob.

harmstyler committed:
Add content/copy support to eZPlatformSearch

This change modifies the addObject function to test if the content object being
added to the solr index is available in the database before continuing.
The legacy module content/copy triggers the addObject function before
the new content is actually in the database, causing an uncaught
exception in the search plugin. The try/catch approach seems best
because the function is not sent any information about how the
contentObject was created. The copied content is then added at a later
time by a frequent cronjob.

harmstyler commented:
Fixes issue #2

harmstyler committed:
variable name update based on feedback from #3

emodric commented:
This PR has a conflict now that #3 is merged :)

Is there really a need for separate cron script here too? Because you can use already existing `ezpublish_legacy/cronjobs/indexcontent.php`. You just need to use `index_object` pending action.

emodric commented on search/plugins/ezplatformsearch/ezplatformsearch.php:
```diff
@@ -3,6 +3,7 @@
 use eZ\Publish\API\Repository\Repository;
 use eZ\Publish\API\Repository\Values\Content\LocationQuery;
 use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
```
Should be `eZ\Publish\API\Repository\Exceptions\NotFoundException`

harmstyler committed:
Merge branch 'master' into addPlatformContentCopySupport

harmstyler committed:
Remove redundant cronjob and references to it

ezplatformindexcontent.php served mostly the same role as
indexcontent.php does. I also updated the action created by the plugin
to use the same identifier that the indexcontent.php cronjob expects.

harmstyler committed:
Change catch to use api base not found exception

harmstyler commented:
I fixed the merge conflict and removed the redundant cron job. I didn't know about the indexcontent cronjob already existing :). I also updated the catch to use the base not found exception.

emodric commented:
Thanks @harmstyler !
